bundle-workflow/src/git/git_repository.py

```diff
@@ -29,29 +29,38 @@ def __init__(self, url, ref, directory=None):
             os.makedirs(self.dir, exist_ok=False)
 
         # Check out the repository
-        self.execute("git init", True)
-        self.execute(f"git remote add origin {self.url}", True)
-        self.execute(f"git fetch --depth 1 origin {self.ref}", True)
-        self.execute("git checkout FETCH_HEAD", True)
-        self.sha = (
-            subprocess.check_output(["git", "rev-parse", "HEAD"], cwd=self.dir)
-            .decode()
-            .strip()
-        )
+        self.working_subdirectory = working_subdirectory
+        self.execute_silent("git init", self.dir)
```

Comment:
I think self.dir will  always be set to a value earlier in this constructor, so it will never be 'None' here.  In execute_silent you'd never use the working_directory?

`cwd = cwd or self.working_directory`

Reply:
That’s what we want. The root dir is where we’re are checking files out (internal to the class), but the working directory is where we run commands from outside.

Reply:
You can call execute_silent externally and it will use working directory by default like all execute calls.

Reply:
ahh I see, thanks!